pkg/events/event_store_test.go

```diff
@@ -55,6 +56,16 @@ func TestStoreAndRetrieve(t *testing.T) {
 	assert.DeepEqual(t, records[0], event1, cmpopts.IgnoreUnexported(si.EventRecord{}))
 	assert.DeepEqual(t, records[1], event2, cmpopts.IgnoreUnexported(si.EventRecord{}))
 
+	// ensure that the underlying array of the return slice of CollectEvents() isn't the same as the one in EventStore.events
```

Comment:
Maybe we can add more records after calling `CollectEvents`, and then we do following checks

1.  the two slices have different points
2.  the elements of first slice from `CollectEvents` are not changed after we put more records.

with those checks we don't need to expose a internal method

Reply:
Change to another version, I think we can directly access the internal event in the test.

Reply:
Change to the third version that accesses store.events directly in the test.